gtsam/navigation/Gal3.cpp

```diff
@@ -0,0 +1,645 @@
+/* ----------------------------------------------------------------------------
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+ * See LICENSE for the license information
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file    Gal3.cpp
+ * @brief   Implementation of 3D Galilean Group SGal(3) state
+ * @authors Matt Kielo, Scott Baker, Frank Dellaert
+ * @date    April 30, 2025 
+ */
+
+#include <gtsam/navigation/Gal3.h> // Note: Adjust path if needed
+#include <gtsam/geometry/SO3.h> // For so3::DexpFunctor if needed, and Rot3 Log/Exp
+#include <gtsam/base/numericalDerivative.h> // For derivative checking if desired
+#include <gtsam/base/Matrix.h> // For Z_3x3 etc.
+#include <gtsam/nonlinear/expressions.h> // For constructing expressions
+
+#include <iostream>
+#include <cmath>
+#include <functional> // For std::function
+
+namespace gtsam {
+
+//------------------------------------------------------------------------------
+// Constants and Helper function for Expmap/Logmap (Implementation Detail)
+//------------------------------------------------------------------------------
+namespace { // Anonymous namespace for internal linkage
+
+  // Define the threshold locally within the cpp file
+  constexpr double kSmallAngleThreshold = 1e-10;
+
+  Matrix3 Calculate_E(const Vector3& theta) {
+      using std::cos;
+      using std::sin;
+      using std::sqrt;
+
+      const double angle_sq = theta.squaredNorm();
+      const Matrix3 W = skewSymmetric(theta);
+      const Matrix3 W2 = W * W;
+
+      Matrix3 E = 0.5 * Matrix3::Identity();
+
+      // Small angle approximation - Use locally defined constant
+      if (angle_sq < kSmallAngleThreshold * kSmallAngleThreshold) {
+          E += (1.0 / 6.0) * W + (1.0 / 24.0) * W2;
+          return E;
+      }
+
+      const double angle = sqrt(angle_sq);
+      const double sin_angle = sin(angle);
+      const double cos_angle = cos(angle);
+
+      // Coefficients from manif::SGal3TangentBase::exp()
+      const double A = (angle - sin_angle) / (angle_sq * angle);                 // A = (theta - sin(theta)) / theta^3
```

Comment:
Inherit from DexpFunctor and make sure not to do double work. Understand the Taylor expansion mechanism in constructor.